src/fireedge/src/modules/components/Forms/VmTemplate/CreateForm/Steps/ExtraConfiguration/inputOutput/graphicsSchema.js

```diff
@@ -0,0 +1,250 @@
+  hu: T.Hungarian,
+  is: T.Icelandic,
+  it: T.Italian,
+  ja: T.Japanese,
+  lv: T.Latvian,
+  lt: T.Lithuanian,
+  mk: T.Macedonian,
+  no: T.Norwegian,
+  pl: T.Polish,
+  pt: T.Portuguese,
+  'pt-br': T.PortugueseBr,
+  ru: T.Russian,
+  sl: T.Slovenian,
+  es: T.SpanishEs,
+  'ca-es': T.SpanishCatalan,
+  sv: T.Swedish,
+  th: T.Thai,
+  tr: T.Turkish,
+  custom: T.Custom,
+}
+
+/** @type {Field} Type field */
+export const TYPE = (isUpdate) => ({
+  name: 'GRAPHICS.TYPE',
+  type: INPUT_TYPES.SWITCH,
+  label: T.Vnc,
+  dependOf: ['HYPERVISOR', '$general.HYPERVISOR'],
+  validation: mixed()
+    .default(() => (isUpdate ? undefined : true))
+    .afterSubmit((value, { context }) => (value ? 'VNC' : undefined))
+    .test('is-valid-type', 'Invalid value', function (value) {
+      if (
+        typeof value === 'boolean' ||
+        value?.toUpperCase() === 'VNC' ||
+        value === undefined
+      ) {
+        return true
+      }
+
+      return false
+    }),
+
+  grid: { md: 12 },
+})
+
+/** @type {Field} Listen field */
+export const LISTEN = (isUpdate) => ({
+  name: 'GRAPHICS.LISTEN',
+  label: T.ListenOnIp,
+  type: INPUT_TYPES.TEXT,
+  dependOf: TYPE().name,
+  htmlType: (noneType) => !noneType && INPUT_TYPES.HIDDEN,
+  validation: string()
+    .trim()
+    .notRequired()
+    .default(() => (isUpdate ? undefined : '0.0.0.0')),
+  fieldProps: { placeholder: '0.0.0.0' },
+  grid: { md: 12 },
+})
+
+/** @type {Field} Port field */
+export const PORT = {
+  name: 'GRAPHICS.PORT',
+  label: T.ServerPort,
+  tooltip: T.ServerPortConcept,
+  type: INPUT_TYPES.TEXT,
+  dependOf: TYPE().name,
+  htmlType: (noneType) => !noneType && INPUT_TYPES.HIDDEN,
+  validation: string()
+    .trim()
+    .notRequired()
+    .default(() => undefined),
+}
+
+/** @type {Field} Keymap field */
+export const KEYMAP = {
+  name: 'GRAPHICS.KEYMAP',
+  label: T.Keymap,
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  dependOf: TYPE().name,
+  values: arrayToOptions(Object.entries(KEYMAP_VALUES), {
+    addEmpty: false,
+    getText: ([_, label]) => label,
+    getValue: ([keymap]) => keymap,
+  }),
+  htmlType: (noneType) => !noneType && INPUT_TYPES.HIDDEN,
+  validation: string()
+    .trim()
+    .notRequired()
+    .transform((value) =>
+      value && KEYMAP_VALUES[value] ? value : CUSTOM_KEYMAP_VALUE
+    )
+    .default(() => undefined)
+    .afterSubmit((value, { context }) =>
+      value === CUSTOM_KEYMAP_VALUE
+        ? context.extra.GRAPHICS.CUSTOM_KEYMAP
+        : value
+    ),
+}
+
+/** @type {Field} Custom keymap field */
+export const CUSTOM_KEYMAP = {
+  name: 'GRAPHICS.CUSTOM_KEYMAP',
+  label: T.Keymap,
+  type: INPUT_TYPES.TEXT,
+  dependOf: KEYMAP.name,
+  htmlType: (selectedKeymap) =>
+    (!selectedKeymap ||
+      selectedKeymap?.toLowerCase() !== CUSTOM_KEYMAP_VALUE) &&
+    INPUT_TYPES.HIDDEN,
+  validation: lazy((_, { context }) =>
+    string()
+      .trim()
+      .when(`$extra.${KEYMAP.name}`, (keymap, schema) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).